activesupport/lib/active_support/callbacks.rb

```diff
@@ -79,14 +79,10 @@ module Callbacks
     #     save
     #   end
     def run_callbacks(kind, &block)
-      send "_run_#{kind}_callbacks", &block
-    end
-
-    private
+      callbacks = send("_#{kind}_callbacks")
 
-    def _run_callbacks(callbacks, &block)
       if callbacks.empty?
-        block.call if block
+        yield if block_given?
```

Comment:
Previously it was optimized to use `block.call`. However, the benchmark script used in https://github.com/rails/rails/pull/17093 did not actually pass a block to the method being called. It was just calling the method with no block.

Stealing benchmark results from https://www.omniref.com/ruby/2.2.0/symbols/Proc/yield#annotation=4088660&line=723, I believe we should be using yield.